### ISSUE TITLE
Test and fix ExtensionResource

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi-aws-native#1405]
- Add support for updates.
- Add integration tests to prevent regressions.

Currently panics and crashes the provider during Deletion:

```
 -  aws-native:index:ExtensionResource log-test deleting (1s) error: error reading from server: EOF
 -  aws-native:index:ExtensionResource log-test **deleting failed** error: error reading from server: EOF
    pulumi:pulumi:Stack untyped-py-p-it-daniels-ma-untyped-py-825813c3  error: update failed
    pulumi:pulumi:Stack untyped-py-p-it-daniels-ma-untyped-py-825813c3 **failed** 1 error; 22 messages
Diagnostics:
  aws-native:index:ExtensionResource (log-test):
    error: error reading from server: EOF

  pulumi:pulumi:Stack (untyped-py-p-it-daniels-ma-untyped-py-825813c3):
    panic: interface conversion: interface {} is resource.PropertyMap, not *resource.Secret
    goroutine 65 [running]:
    github.com/pulumi/pulumi/sdk/v3/go/common/resource.PropertyValue.SecretValue(...)
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.107.0/go/common/resource/properties.go:460
    github.com/pulumi/pulumi-aws-native/provider/pkg/provider.parseCheckpointObject(...)
        /Users/dan/Dev/pulumi-aws-native/provider/pkg/provider/provider.go:1378
    github.com/pulumi/pulumi-aws-native/provider/pkg/provider.(*cfnProvider).Delete(0x14002710a20, {0x1054ac770, 0x140025ff980}, 0x14000976000)
        /Users/dan/Dev/pulumi-aws-native/provider/pkg/provider/provider.go:1171 +0x7f0
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Delete_Handler.func1({0x1054ac770, 0x140025ff980}, {0x10532d6c0?, 0x14000976000})
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.107.0/proto/go/provider_grpc.pb.go:647 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x1054ac770, 0x140025ff410}, {0x10532d6c0, 0x14000976000}, 0x14000e077c0, 0x140002b9770)
        /Users/dan/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x2e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Delete_Handler({0x10542bbe0?, 0x14002710a20}, {0x1054ac770, 0x140025ff410}, 0x14000d94400, 0x14000d92ce0)
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.107.0/proto/go/provider_grpc.pb.go:649 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x140003c43c0, {0x1054ac770, 0x140025ff380}, {0x1054b9e20, 0x14000190d00}, 0x1400261e900, 0x14002603d40, 0x106826ca8, 0x0)
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1343 +0xb8c
    google.golang.org/grpc.(*Server).handleStream(0x140003c43c0, {0x1054b9e20, 0x14000190d00}, 0x1400261e900)
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:1737 +0x990
    google.golang.org/grpc.(*Server).serveStreams.func1.1()
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:986 +0x88
    created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 25
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.59.0/server.go:997 +0x160

    error: update failed
```
